packages/utils/src/lib/git.ts

```diff
@@ -17,3 +17,40 @@ export async function getLatestCommit() {
   });
   return log.latest;
 }
+
+export async function branchHasChanges(): Promise<boolean> {
+  return await git.status(['-s']).then(r => r.files.length > 0);
+}
+
+export async function guardAgainstDirtyRepo(): Promise<void> {
+  const isDirty = await branchHasChanges();
+  if (isDirty) {
+    throw new Error(`
+        Repository should be clean before we you can proceed.
+        Commit your local changes or stash them.
+      `);
+  }
+}
+
+export async function getCurrentBranchOrTag(): Promise<string> {
+  return (
+    (await git.branch().then(r => r.current)) ||
+    // @TODO replace with simple git
+    (await git.raw(['describe --tags --exact-match']).then(out => out.trim()))
+  );
+}
+
+export async function safeCheckout(
+  branchOrHash: string,
+  options: {
+    clean?: boolean;
+  } = {},
+): Promise<void> {
+  // git requires a clean history to check out a branch
+  if (options?.clean) {
+    await git.clean(['f']);
+    console.info(`branch cleaned`);
+  }
```

Comment:
Is it necessary to remove untracked files?

Reply:
The repo is not clean otherwise. and it is optional.
I use it in the history command to revert repo changes.